microservices/first/utils/db_manager.py

```diff
@@ -123,10 +127,11 @@ def update_deployment_fields(
         :param deployment_id: The ID of the deployment to update.
         :param updates: A dictionary where keys are column names and values are
             the new values for those columns.
+        :throws KeyError if `updates` is missing necessary field
         """
-        parameters = [f"{key} = ?" for key in updates.keys()]
+        parameters = [f"{key} = ?" for key in DEPLOYMENT_FIELDS]
         sql = f"UPDATE deployments SET {', '.join(parameters)} WHERE id = ?"
-        values = list(updates.values()) + [deployment_id]
+        values = list(updates[key] for key in DEPLOYMENT_FIELDS) + [deployment_id]
 
```

Comment:
This has been changed from only updating given fields to updating all deployment fields. The leader sends only the `status` in its updates rather than all the fields. This is because the leader while in its lifecycle won't have a different `repo_name` nor `repo_branch`.

I think this should be changed from `DEPLOYMENT_FIELDS` back to only the fields given in `updates`